src/store_client.cc

```diff
@@ -527,7 +533,7 @@ store_client::readBody(const char *, ssize_t len)
 
     if (len > 0 && rep && entry->mem_obj->inmem_lo == 0 && entry->objectLen() <= (int64_t)Config.Store.maxInMemObjSize && Config.onoff.memory_cache_disk) {
         storeGetMemSpace(len);
-        // The above may start to free our object so we need to check again
+        // recheck for the above call may purge entry's data from the memory cache
```

Comment:
The old comment was rather misleading: It is not possible to _start_ to free a C/C++ object, so the comment was probably referring to cached data rather than the StoreEntry object itself. The current official storeGetMemSpace() code does not really destroy any locked StoreEntries, and the official code below assumes that `entry` object is still alive (or we would not be able to "check again" anything!). Thus, again, the original comment was probably talking about cached data.

The current storeGetMemSpace() implementation does not purge cached data of locked entries either, but there is no guarantee that it would not start doing that in the future, so the second check is (overall) a legitimate idea even now, when we can finally rely on the `entry` being properly locked.